### PR TITLE
Setup Smile Config during Runtime

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/9126/allow-setting-smlie-config-at-runtime

## Summary

This PR allows a user to set the Smile Config at runtime, if no `smile_config.json` file is supplied

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

Screenshot_20230713_180854
Screenshot_20230713_180923

### CODE REVIEW

File: sample/src/main/java/com/smileidentity/sample/repo/DataStoreRepository.kt [hunk 1]
```diff
@@ -70,6 +70,16 @@ object DataStoreRepository {
...
     }
 
+    /**
+     * Get the Smile string Config from [mainDataStore], if one has been manually set. (It may not be set if
+     * the sample app is utilizing the assets/smile_config.json file instead).
+     *
+     * If no config is set, this returns null
+     */
+    fun getStringConfig(): Flow<String?> = mainDataStore.data.map { preferences ->
+        preferences[Keys.config] ?: ""
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/repo/DataStoreRepository.kt:80]: Why return an empty String here? That would not conform to Config json schema

```suggestion
        preferences[Keys.config]
```

File: sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt [hunk 2]
```diff
@@ -160,8 +163,8 @@ class MainScreenViewModel : ViewModel() {
      * will automatically pick up the correct environment)
      */
     fun toggleEnvironment() {
-        pendingJobCountJob.cancel()
-        backgroundJobsPollingJob.cancel()
+        pendingJobCountJob?.cancel()
+        backgroundJobsPollingJob?.cancel()
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt:167]: These are redundant, they are never null
```suggestion
        pendingJobCountJob.cancel()
        backgroundJobsPollingJob.cancel()
```

File: sample/src/main/java/com/smileidentity/sample/viewmodel/RootViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        )
+
+    private val configAdapter = SmileID.moshi.adapter(Config::class.java)
+    private val currentConfig = DataStoreRepository.getConfig()
+        .map { it?.let { configAdapter.toJson(it) } ?: "Smile Config from the Smile ID Portal" }
+        .stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.WhileSubscribed(),
+            initialValue = "",
+        )
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/viewmodel/RootViewModel.kt:49]: This can be replaced with getStringConfig now

File: sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt [hunk 4]
```diff
@@ -171,6 +174,20 @@ class MainScreenViewModel : ViewModel() {
...
+            initialValue = "",
+        )
+
+    val smileConfig = DataStoreRepository.getConfig()
+        .stateIn(
+            scope = viewModelScope,
+            started = Eagerly,
+            initialValue = null,
+        )
+
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/viewmodel/MainScreenViewModel.kt:190]: These seem to be unused
```suggestion
```

File: sample/src/main/java/com/smileidentity/sample/viewmodel/SettingsViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    val smileConfigHint: String = "Your Smile Config from the Smile ID Portal",
+    @StringRes val smileConfigError: Int? = null,
+)
+
+class SettingsViewModel : ViewModel() {
+    private val _uiState = MutableStateFlow(SettingsUiState())
+    val uiState = _uiState.asStateFlow()
+    private val configAdapter = SmileID.moshi.adapter(Config::class.java)
+    private val currentConfig = DataStoreRepository.getConfig()
+        .map { it?.let { configAdapter.toJson(it) } ?: "Smile Config from the Smile ID Portal" }
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/viewmodel/SettingsViewModel.kt:29]:
```suggestion
    private val currentConfig = DataStoreRepository.getStringConfig()
        .map { it ?: "Smile Config from the Smile ID Portal" }
```

File: sample/src/main/java/com/smileidentity/sample/repo/DataStoreRepository.kt [hunk 6]
```diff
@@ -70,6 +70,16 @@ object DataStoreRepository {
         mainDataStore.edit { it.clear() }
     }
 
+    /**
+     * Get the Smile string Config from [mainDataStore], if one has been manually set. (It may not be set if
+     * the sample app is utilizing the assets/smile_config.json file instead).
+     *
+     * If no config is set, this returns null
+     */
+    fun getStringConfig(): Flow<String?> = mainDataStore.data.map { preferences ->
```
Comment (vanshg) [inline sample/src/main/java/com/smileidentity/sample/repo/DataStoreRepository.kt:79]: Thoughts on renaming to `getConfigJsonString`? for 2 reasons: 
- so that it shows up for autocomplete when we start typing `getC`...
- To be explicit that this is not the `toString` of `Config` but the json encoding

**Review by vanshg — APPROVED** [review]
